src/main/java/org/candlepin/service/model/CloudCheckInEvent.java

```diff
@@ -0,0 +1,219 @@
+        this.cloudProviderId = cloudProviderId;
+        this.cloudAccountId = cloudAccountId;
+        this.cloudOfferingIds = cloudData.getCloudOfferingIds();
+
+        try {
+            eventBody = getSerializedBody();
+        }
+        catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * @return the UUID of the consumer that has had an update to the last check-in date
+     */
+    public String getConsumerUuid() {
+        return consumerUuid;
+    }
+
+    /**
+     * @return the check-in date for this event
+     */
+    public Date getCheckIn() {
+        return checkIn;
+    }
+
+    /**
+     * @return the cloud provider's short name that the consumer belongs to
+     */
+    public String getCloudProviderId() {
+        return cloudProviderId;
+    }
+
+    /**
+     * @return the cloud account ID that the consumer belongs to
+     */
+    public String getCloudAccountId() {
+        return cloudAccountId;
+    }
+
+    /**
+     * @return the cloud offerings for the consumer that has had an update to the last check-in date
+     */
+    public List<String> getCloudOfferingIds() {
+        return cloudOfferingIds;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getBody() {
+        return eventBody;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SerializationType getSerializationType() {
+        return SerializationType.JSON;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EventType getEventType() {
+        return EventType.CLOUD_SYSTEM_CHECK_IN;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (!(obj instanceof CloudCheckInEvent)) {
+            return false;
+        }
+
+        CloudCheckInEvent that = (CloudCheckInEvent) obj;
+
+        EqualsBuilder builder = new EqualsBuilder()
+            .append(this.getConsumerUuid(), that.getConsumerUuid())
+            .append(this.getCheckIn(), that.getCheckIn())
+            .append(this.getCloudProviderId(), that.getCloudProviderId())
+            .append(this.getCloudAccountId(), that.getCloudAccountId())
+            .append(this.getCloudOfferingIds(), that.getCloudOfferingIds());
+
+        return builder.isEquals();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        HashCodeBuilder builder = new HashCodeBuilder(37, 7)
+            .append(this.getConsumerUuid())
+            .append(this.getCheckIn())
+            .append(this.getCloudProviderId())
+            .append(this.getCloudAccountId())
+            .append(this.getCloudOfferingIds());
+
+        return builder.toHashCode();
+    }
+
+    private String getSerializedBody() throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
ObjectMapper instance creation is expensive, which is why we've limited the amount of instances we create by centralizing it to ObjectMapperFactory.java, and then marking each as a Singleton and Named in CandlepinModule.java. I think you can just re-use one of the existing instances by injecting it in this class (HypervisorUpdateJobObjectMapper looks the best, since it is just a `new Objectmapper()`  with the only configuration applied to it being on the deserialization side, which we don't care about in this case).

Reply:
That's a good point, although I don't think we can directly inject the mapper into the `CloudCheckInEvent` class since there can be multiple instances of the class and I think you would have to use the Guice Injector to create an instance of the class that way. 

I did add the ObjectMapper as a mandatory argument to the constructor, so the class that is creating the CloudCheckInEvents can inject the ObjectMapper into that class as it is probably a manager or resource class, and then pass the ObjectMapper to the CloudCheckInEvent.